Social: Add the showNudge reducer

pablinos opened the pull request:
The reducer for showNudge was missing, so it would default back to `false` if a value in the store changed. This meant that after clicking 'Start for Free' on the pricing page, it would seem like the user had a paid plan.

#### Changes proposed in this Pull Request:
This PR adds the missing resolver.

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Set up a Jetpack Social site and enable sharing limits
* Click the 'Start for Free' option on the pricing page after connecting the site.
* With this change you'll see the shares meter for the free plan. Without the change you'll see the stats cards, that we show for the paid plan, or if the limits aren't enabled.

pablinos committed:
Social: Add the showNudge reducer

The reducer for showNudge was missing, so it would default back to
`false` if a value in the store changed. This meant that after clicking
'Start for Free' on the pricing page, it would seem like the user had a
paid plan.

danielpost commented on projects/plugins/social/src/js/store/reducer/index.js:
```diff
@@ -9,6 +9,7 @@ const reducer = combineReducers( {
 	siteData,
 	connectionData,
 	jetpackSettings,
+	showNudge: ( state = true ) => state,
```
Rather than defining it separately and explicitly like this, should we just move the `showNudge` param to the `jetpackSettings`? Similar to `showPricingPage` and `hasPaidPlan`.

pablinos replied:
Yes, I was thinking of doing that, and then thought it was too big a change for this stage, but actually it's probably not. `hasPaidPlan` is already a `jetpackSettings` selector. 

We should probably reverse the logic at some point anyway.

danielpost reviewed (dismissed):
Works well! I added a comment but we can address that in a follow-up as well.

pablinos committed:
Move showNudge to jetpackSettings in the store.

pablinos commented:
> Works well! I added a comment but we can address that in a follow-up as well.

Too late, I updated it. What do you think of the version in [f8f17a3](https://github.com/Automattic/jetpack/pull/26635/commits/f8f17a39df35ff5192b04d92f5e9ac5d640da3e6) @danielpost?

pablinos committed:
Remove the default.

pablinos committed:
Add a default in case showNudge isn't present for any reason.
